base/applications/mscutils/eventvwr/eventvwr.c

```diff
@@ -3807,6 +3807,36 @@ WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
             break;
         }
 
+        case WM_CONTEXTMENU:
+        {
+            RECT rc;
+            HTREEITEM hItem;
+            TVHITTESTINFO hInfo = {0};
+
+            INT xPos = GET_X_LPARAM(lParam);
+            INT yPos = GET_Y_LPARAM(lParam);
+
+            GetWindowRect(hwndTreeView, &rc);
+            hInfo.pt.x = xPos - rc.left;
+            hInfo.pt.y = yPos - rc.top;
+
+            if ((hItem = TreeView_HitTest(hwndTreeView, &hInfo)))
+            {
+                TreeView_SelectItem(hwndTreeView, hItem);
+
+                if (TreeView_GetParent(hwndTreeView, hItem))
+                {
+                    HMENU hCtxtMenuBar = LoadMenu(hInst, MAKEINTRESOURCE(IDM_EVENTWR_CTX));
+                    HMENU contextMenu = GetSubMenu(hCtxtMenuBar, 0);
+
+                    SendMessage(hWnd, WM_COMMAND, TrackPopupMenuEx(contextMenu,
+                                TPM_LEFTALIGN|TPM_TOPALIGN|TPM_NONOTIFY|TPM_RETURNCMD,
+                                xPos, yPos, hWnd, NULL), (LPARAM)hwndTreeView);
```

Comment:
Why isn't `TrackPopupMenuEx()` all by itself (without the SendMessage stuff etc.) sufficient?